x-pack/plugins/ml/public/data_frame/components/define_pivot/common.ts

```diff
@@ -29,33 +30,39 @@ export interface Field {
 
 function getDefaultGroupByConfig(
   aggName: string,
+  dropDownName: string,
   fieldName: FieldName,
   groupByAgg: PIVOT_SUPPORTED_GROUP_BY_AGGS
-) {
+): PivotGroupByConfig {
   switch (groupByAgg) {
     case PIVOT_SUPPORTED_GROUP_BY_AGGS.TERMS:
       return {
         agg: groupByAgg,
         aggName,
+        dropDownName,
         field: fieldName,
       };
     case PIVOT_SUPPORTED_GROUP_BY_AGGS.HISTOGRAM:
       return {
         agg: groupByAgg,
         aggName,
+        dropDownName,
         field: fieldName,
         interval: '10',
       };
     case PIVOT_SUPPORTED_GROUP_BY_AGGS.DATE_HISTOGRAM:
       return {
         agg: groupByAgg,
         aggName,
+        dropDownName,
         field: fieldName,
         interval: '1m',
       };
   }
 }
 
+const illegalEsAggNameChars = /[[\]>]/g;
```

Comment:
i would have thought there are a lot more illegal characters for an aggregation name, is there something significant about these ones?

Reply:
I was surprised too. Originally it was `[a-zA-Z0-9\\-_]`, here's the reference to the elasticsearch PR that changed this: https://github.com/elastic/elasticsearch/pull/6708